gulpfile.js

```diff
@@ -297,6 +303,224 @@ gulp.task('minifyRelease', ['generateStubs'], function() {
+
+    var iface = readline.createInterface({
+        input: process.stdin,
+        output: process.stdout
+    });
+
+    if (argv.confirm) {
+        // skip prompt for travis
+        return deployCesium(bucketName, uploadDirectory, cacheControl);
+    }
+
+    // prompt for confirmation
+    iface.question('Files from your computer will be published to the ' + bucketName + ' bucket. Continue? [y/n] ', function(answer) {
+        iface.close();
+        if (answer === 'y') {
+            deployCesium(bucketName, uploadDirectory).then(function() {
+                done();
+            });
+        } else {
+            console.log('Deploy aborted by user.');
+            done();
+        }
+    });
+
+});
+
+// Deploy cesium to aws
+function deployCesium(bucketName, uploadDirectory, cacheControl) {
+    var readFile = Promise.promisify(fs.readFile);
+    var gzip = Promise.promisify(zlib.gzip);
+    var concurrencyLimit = 2000;
+
+    var s3 = new Promise.promisifyAll(new aws.S3({
+        maxRetries : 10,
+        retryDelayOptions : {
+            base : 500
+        }
+    }));
+
+    var existingBlobs = [];
+    var totalFiles = 0;
+    var uploaded = 0;
+    var skipped = 0;
+
+    return listAll(s3, bucketName, uploadDirectory, existingBlobs)
+    .then(function() {
+        return globby([
+            'Apps/**',
+            'Build/**',
+            'Source/**',
+            'Specs/**',
+            'ThirdParty/**',
+            '*.md',
+            'favicon.ico',
+            'gulpfile.js',
+            'index.html',
+            'logo.png',
+            'package.json',
+            'server.js',
+            'web.config'
+        ], {
+            dot : true, // include hidden files
+            nodir : true // only directory files
+        });
+    }).then(function(files) {
+        return Promise.map(files, function(file) {
+            var blobName = uploadDirectory + '/' + file;
+            var mimeLookup = getMimeType(blobName);
+            var contentType = mimeLookup.type;
+            var compress = mimeLookup.compress;
+            var contentEncoding = compress ? 'gzip' : undefined;
+            var contents;
+            var etag;
+
+            totalFiles++;
+
+            return readFile(file)
+                .then(function(content) {
+                    if (compress) {
+                        return gzip(content).then(function(result) {
+                            contents = result;
+                        });
+                    } else {
+                        contents = content;
+                    }
+                })
+                .then(function() {
+                    // compute hash and etag
+                    var hash = crypto.createHash('md5').update(contents).digest('hex');
+                    etag = crypto.createHash('md5').update(contents).digest('base64');
+
+                    var index = existingBlobs.indexOf(blobName);
+                    if (index > -1) {
```

Comment:
Always favor early return, that is, check for `index <= -1` and return true immediately instead of having a big chunk of code in the if block.